tests/app/submitter/test_submitter.py

```diff
@@ -1,276 +1,251 @@
-            published = submitter.send_message(
-                message={"test_data"},
-                tx_id="123",
-                case_id="456",
-            )
-
-            # Then
-            bucket = client.return_value.get_bucket.return_value
-            blob = bucket.blob.return_value
-            assert isinstance(blob.metadata, dict)
-
-            blob_name = bucket.blob.call_args[0][0]
-            assert blob_name == "123"
-
-            blob_contents = blob.upload_from_string.call_args[0][0]
-            assert blob_contents == b"{'test_data'}"
-
-            assert published is True
-
-    @staticmethod
-    def test_send_message_adds_metadata():
-        with patch("app.submitter.submitter.storage.Client") as client:
-            # Given
-            submitter = GCSSubmitter(bucket_name="test_bucket")
-
-            # When
-            submitter.send_message(
-                message={"test_data"},
-                tx_id="123",
-                case_id="456",
-            )
-
-            # Then
-            bucket = client.return_value.get_bucket.return_value
-            blob = bucket.blob.return_value
-
-            assert blob.metadata == {
-                "tx_id": "123",
-                "case_id": "456",
-            }
-
-
-class TestGCSFeedbackSubmitter(TestCase):
-    @staticmethod
-    @patch("app.submitter.submitter.storage.Client")
-    def test_upload_feedback(client):
-        # Given
-        feedback = GCSFeedbackSubmitter(bucket_name="feedback")
-
-        metadata = {
-            "feedback_count": 1,
-            "feedback_submission_date": "2021-03-23",
-            "form_type": "H",
-            "language_code": "cy",
-            "region_code": "GB-ENG",
-            "tx_id": "12345",
-        }
-
-        payload = {
-            "feedback-type": "Feedback type",
-            "feedback-text": "Feedback text",
-        }
-
-        payload.update(metadata)
-
-        # When
-        feedback_upload = feedback.upload(metadata, json_dumps(payload))
-
-        # Then
-        bucket = client.return_value.get_bucket.return_value
-        blob = bucket.blob.return_value
-
-        assert blob.metadata["feedback_count"] == 1
-        assert blob.metadata["feedback_submission_date"] == "2021-03-23"
-        assert blob.metadata["form_type"] == "H"
-        assert blob.metadata["language_code"] == "cy"
-        assert blob.metadata["tx_id"] == "12345"
-        assert blob.metadata["region_code"] == "GB-ENG"
-
-        blob_contents = blob.upload_from_string.call_args[0][0]
-
-        assert (
-            blob_contents
-            == b'{"feedback-type": "Feedback type", "feedback-text": "Feedback text", '
-            b'"feedback_count": 1, "feedback_submission_date": "2021-03-23", '
-            b'"form_type": "H", "language_code": "cy", "region_code": "GB-ENG", "tx_id": "12345"}'
-        )
-        assert feedback_upload is True
+def test_when_fail_to_connect_to_queue_then_published_false(
```

Comment:
I think classes would still help. Grouping them by submitter makes sense. At the moment, the method names aren't that clear given it isn't grouped by the submitter. That, or maybe separate files?.

Reply:
Well it's not for convenience, it's for readability and clarity. I think prefixing with the submitter solves it.